sources/packages/backend/libs/utilities/src/string-builder.ts

```diff
@@ -83,3 +86,62 @@ export class StringBuilder {
     return this.contents.join("");
   }
 }
+
+/**
+ * Advanced string builder with specialized functions.
+ */
+export class AdvancedStringBuilder extends StringBuilder {
+  private readonly stringFiller: string;
+  private readonly numberFiller: string;
+  private readonly dateFiller: string;
+  constructor(stringFiller: string, numberFiller: string, dateFiller: string) {
```

Comment:
The constructor can be changed as below.
```ts
constructor(
  private readonly stringFiller: string,
  private readonly numberFiller: string,
  private readonly dateFiller: string,
) {
  super();
}
```

Reply:
Still using my java brain 😂